### PR TITLE
fix: Date Picker timezone

### PR DESCRIPTION
**Pull request by Pengap**

### CODE REVIEW

File: packages/sdk/src/components/editor/date/Editor.tsx [hunk 1]
```diff
@@ -28,12 +35,12 @@ const DateEditorBase: ForwardRefRenderFunction<IEditorRef<string>, IDateEditorMa
   const valueComponent = useMemo(() => {
     if (!value) return <span>{t('editor.date.placeholder')}</span>;
 
-    let format = 'YYYY-MM-DD HH:mm';
-    if (date && time) {
+    let format = enableTimePicker ? 'YYYY-MM-DD HH:mm' : 'YYYY-MM-DD';
```
Comment (tea-artist) [inline packages/sdk/src/components/editor/date/Editor.tsx]: Maybe we should use format in dateField options

**Review by tea-artist — APPROVED** [review]

File: packages/sdk/src/components/editor/date/EditorMain.tsx [hunk 2]
```diff
@@ -21,13 +23,16 @@ const DateEditorMainBase: ForwardRefRenderFunction<IEditorRef<string>, IDateEdit
...
-  const { value, style, className, onChange, readonly, options } = props;
+  const { value, style, className, onChange, readonly, options, enableTimePicker } = props;
   const inputRef = useRef<HTMLInputElement | null>(null);
-  const { time } = options?.formatting || {};
+  const { time, timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone } =
+    options?.formatting || {};
   const [date, setDate] = useState<string | null>(value || null);
-  const hasTimePicker = time !== TimeFormatting.None;
+  const hasTimePicker =
+    enableTimePicker !== undefined ? enableTimePicker : time !== TimeFormatting.None;
```
Comment (Sky-FE) [inline packages/sdk/src/components/editor/date/EditorMain.tsx]: To simplify the code, consider the following approaches:
```
const hasTimePicker = enableTimePicker && time !== TimeFormatting.None;
```

File: packages/sdk/src/components/editor/date/EditorMain.tsx [hunk 3]
```diff
@@ -36,48 +41,65 @@ const DateEditorMainBase: ForwardRefRenderFunction<IEditorRef<string>, IDateEdit
...
   const onSelect = (value?: Date) => {
-    let curDatetime = dayjs(value);
-    const prevDatetime = dayjs(date);
-    if (!curDatetime.isValid()) onChange?.(null);
-
-    if (prevDatetime.isValid()) {
-      const hours = prevDatetime.get('hour');
-      const minutes = prevDatetime.get('minute');
-      curDatetime = curDatetime.set('hour', hours).set('minute', minutes);
+    if (!value) onChange?.(null);
```
Comment (Sky-FE) [inline packages/sdk/src/components/editor/date/EditorMain.tsx]: if (!value) return onChange?.(null);

**Review by Sky-FE — APPROVED** [review]

**Comment by tea-artist** [comment]
Thank God the coverage worked!

**Pull request merged**